imports/ui/case-explorer/case-explorer.jsx

```diff
@@ -144,16 +153,38 @@ class CaseExplorer extends Component {
   )
   filterCases = memoizeOne(
     (cases, searchText) => {
+      this.primeCaseId = null
       if (!searchText) return cases
-      let regex
 
-      // Checking the first character of the search term to determine if it's a numerical or alphabetical search
-      if (searchText.charAt(0).match(/\d/)) {
-        regex = new RegExp(`(^|[^\\d])${searchText}`, 'i') // Searching for a number starting as the search term
-      } else {
-        regex = new RegExp(`(^|[^a-zA-Z])${searchText}`, 'i') // Searching for a word starting as the search term
+      let caseResult, results, selectedCfg
```

Comment:
should use `const` where this symbols are used, instead of hoisting definitions with `let`. They're only assigned a value once